geowrangler/grids.py

```diff
@@ -240,14 +246,34 @@ def generate_grid(self: BingTileGridGenerator, gdf: GeoDataFrame) -> DataFrame:
         for geom in reprojected_gdf.unary_union.geoms:
             _tiles = self.get_tiles_for_polygon(geom)
             tiles.update(_tiles)
-    quadkey, geom = zip(*((k, v) for k, v in tiles.items()))
+    quadkey, geom_tile = zip(*((k, v) for k, v in tiles.items()))
+    geom, tile = zip(*geom_tile)
+
+    result = {"quadkey": list(quadkey)}
+
+    if self.add_xyz_cols:
```

Comment:
small nitpick but this can simplified to
```python
    if self.add_xyz_cols:
        result["x"] = [t.x for t in tile]
        result["y"] = [t.y for t in tile]
        result["z"] = [t.z for t in tile]

    if self.return_geometry:
        tiles_df = GeoDataFrame(
            result,
            geometry=list(geom),
            crs="epsg:4326",
        )
        tiles_df = tiles_df.to_crs(gdf.crs)
    else:
        tiles_df = DataFrame(result)

    return tiles_df
```